### PR TITLE
[common library] Add common library cluster helper

### PR DESCRIPTION
**Pull request by alvarocabanas**
Add common library cluster helper

### CODE REVIEW

File: library/common-library/templates/_cluster.tpl [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+Return the cluster
+*/}}
+{{- define "common.cluster" -}}
+    {{- if .Values.cluster -}}
+        {{- .Values.cluster -}}
+    {{- else if .Values.global -}}
+        {{- if .Values.global.cluster -}}
+            {{- .Values.global.cluster -}}
+        {{ else }}
+            {{- "" -}}
```
Comment (roobre) [inline library/common-library/templates/_cluster.tpl:11]: Do we need these `else ""` constructs? In my past experience it was fine to simply omit them, Helm will render nothing by default which should be evaluated to `false`.

File: library/CHART-TEMPLATE/tests/common_images_pull_secrets_test.yaml [hunk 2]
```diff
@@ -4,17 +4,18 @@ templates:
 release:
   name: my-release
   namespace: my-namespace
-set:
 tests:
   - it: does not set imagePullSecrets if not specified
     set:
+      cluster: test-cluster
```
Comment (roobre) [inline library/CHART-TEMPLATE/tests/common_images_pull_secrets_test.yaml:10]: Why do we need to set `cluster: test-cluster` for _all_ test cases?

File: library/common-library/templates/_cluster.tpl (hunk 1 above)
Comment (kang-makes) [reply library/common-library/templates/_cluster.tpl:11]: The helper will return an empty string if you print nothing. This is redundant. No helper has those so I would remove this.

**Review by kang-makes — COMMENTED** [review]
I just pointed a few nits :)

File: library/common-library/templates/_cluster.tpl (hunk 1 above)
Comment (alvarocabanas) [reply library/common-library/templates/_cluster.tpl:11]: oks, I will remove it

File: library/CHART-TEMPLATE/tests/common_images_pull_secrets_test.yaml (hunk 2 above)
Comment (alvarocabanas) [reply library/CHART-TEMPLATE/tests/common_images_pull_secrets_test.yaml]: removed

File: library/common-library/templates/_cluster.tpl (hunk 1 above)
Comment (alvarocabanas) [reply library/common-library/templates/_cluster.tpl]: done

File: library/CHART-TEMPLATE/tests/common_cluster_test.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+release:
+  name: my-release
+  namespace: my-namespace
+set:
+tests:
+  - it: sets null as cluster name if not specified
+    asserts:
+      - equal:
+          path: spec.template.spec.containers[0].env[0].value
+          value: null
```
[the remark this answers is not in the document]
Comment (alvarocabanas) [reply library/CHART-TEMPLATE/tests/common_cluster_test.yaml:13]: done

File: library/common-library/templates/_cluster.tpl [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+Return the cluster
+*/}}
+{{- define "common.cluster" -}}
+    {{- if .Values.cluster -}}
+        {{- .Values.cluster -}}
+    {{- else if .Values.global -}}
+        {{- if .Values.global.cluster -}}
+            {{- .Values.global.cluster -}}
+        {{- end -}}
+    {{- end -}}
```
Comment (paologallinaharbur) [inline library/common-library/templates/_cluster.tpl:11]: I believe we discussed this already, therefore I am just double-checking.
We decided not to fail here and to leave the decision to each chart, rigt?

**Review by paologallinaharbur — APPROVED** [review]
Just a small doubt, everything else looks good!

File: library/common-library/templates/_cluster.tpl (hunk 4 above)
Comment (alvarocabanas) [reply library/common-library/templates/_cluster.tpl:11]: yes

**Pull request merged**